packages/ng/core-select/panel/key-manager.ts

```diff
@@ -0,0 +1,103 @@
+import { ActiveDescendantKeyManager } from '@angular/cdk/a11y';
+import { DestroyRef, EventEmitter, Injectable, QueryList, inject } from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { Observable, asyncScheduler, debounceTime, filter, map, observeOn, take } from 'rxjs';
+import { ɵLuOptionComponent } from '../option';
+
+interface CoreSelectKeyManagerOptions<T> {
+	queryList: QueryList<ɵLuOptionComponent<T>>;
+	options$: Observable<T[]>;
+	optionComparer: (a: T, b: T) => boolean;
+	activeOptionIdChanged$: EventEmitter<string>;
+	clueChange$: Observable<string>;
+}
+
+@Injectable()
+export class CoreSelectKeyManager<T> {
+	#destroyRef = inject(DestroyRef);
+	#cdkKeyManager?: ActiveDescendantKeyManager<ɵLuOptionComponent<T>>;
+	#hasSearchChanged = false;
+	#options?: CoreSelectKeyManagerOptions<T>;
+
+	init(options: CoreSelectKeyManagerOptions<T>): void {
+		this.#options = options;
+		this.#cdkKeyManager = new ActiveDescendantKeyManager(options.queryList).withHomeAndEnd();
+		this.#bindClueChange(options.clueChange$);
+		this.#bindOptionsChange(options);
+		this.#bindActiveOptionIdChanged(options.activeOptionIdChanged$);
+		this.#cdkKeyManager.setFirstItemActive();
+	}
+
+	onKeydown(event: KeyboardEvent): void {
+		this.#cdkKeyManager?.onKeydown(event);
+	}
+
+	get activeItem(): ɵLuOptionComponent<T> | undefined {
+		return this.#cdkKeyManager?.activeItem;
+	}
+
+	get activeItemIndex(): number {
+		return this.#cdkKeyManager?.activeItemIndex ?? -1;
+	}
+
+	highlightOption(option: T): void {
+		if (!this.#options) {
+			return;
+		}
+
+		const { options$, optionComparer } = this.#options;
+
+		options$
+			.pipe(
+				observeOn(asyncScheduler),
+				map((options) => options.findIndex((o) => optionComparer(o, option))),
+				filter((index) => index !== -1),
+				take(1),
+				takeUntilDestroyed(this.#destroyRef),
```

Comment:
Isn't `take(1)` sufficient here? This is subscribed when calling a function so I guess we want to destroy it asap?
```suggestion
				take(1),
```

Reply:
If the observable never emits or after the component is destroyed, the Subscription would live for ever / too long